src/functions/sendCandidateResults/application/service/categories/AM2/fault-provider-cat-a-mod2.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  TestData as CatAMod2TestData,
+  SafetyAndBalanceQuestions,
+} from '@dvsa/mes-test-schema/categories/AM2';
+import { convertNumericFaultObjectToArray, convertBooleanFaultObjectToArray } from '../../fault-provider';
+import { Fault } from '../../../../domain/fault';
+import { CompetencyOutcome } from '../../../../domain/competency-outcome';
+import { Competencies } from '../../../../domain/competencies';
+
+export const getDrivingFaultsCatAMod2 = (testData: CatAMod2TestData | undefined): Fault[] => {
+
+  if (!testData) {
+    throw new Error('No Test Data');
+  }
+
+  return [
+    ...convertNumericFaultObjectToArray(testData.drivingFaults),
+    ...getSafetyAndBalanceFaultCatAMod2(testData.safetyAndBalanceQuestions),
+  ];
+};
+
+export const getSeriousFaultsCatAMod2 = (testData: CatAMod2TestData | undefined): Fault[] => {
+
+  if (!testData) {
+    throw new Error('No Test Data');
+  }
+
+  return [
+    ...convertBooleanFaultObjectToArray(testData.seriousFaults),
+  ];
+};
+
+export const getDangerousFaultsCatAMod2 = (testData: CatAMod2TestData | undefined): Fault[] => {
+
+  if (!testData) {
+    throw new Error('No Test Data');
+  }
+
+  return [
+    ...convertBooleanFaultObjectToArray(testData.dangerousFaults),
+  ];
+};
+
+export const getSafetyAndBalanceFaultCatAMod2 = (
+  safetyAndBalanceQuestions: SafetyAndBalanceQuestions | undefined,
+): Fault[] => {
+  const faultArray: Fault[] = [];
+
+  if (hasQuestionFault(safetyAndBalanceQuestions)) {
+    faultArray.push(
+      { name: Competencies.safetyAndBalance, count: 1 },
+    );
+  }
+  return faultArray;
+};
+
+export const hasQuestionFault = (
```

Comment:
This function always returns `false`

Reply:
I have made some changes to this method.

Reply:
I would use `Array.prototype.some()` function instead of `forEach()` in this scenario. However the functionality is correct now